measure/src/main/scala/org/apache/griffin/measure/configuration/dqdefinition/DQConfig.scala

```diff
@@ -154,8 +154,8 @@ case class RuleParam(@JsonProperty("dsl.type") private val dslType: String,
                      @JsonProperty("out") private val outputs: List[RuleOutputParam] = null,
                      @JsonProperty("error.confs") private val errorConfs: List[RuleErrorConfParam] = null
                     ) extends Param {
-  def getDslType: DslType = if (dslType != null) DslType(dslType) else DslType("")
-  def getDqType: DqType = if (dqType != null) DqType(dqType) else DqType("")
+  def getDslType: DslType.DslType = if (dslType != null) DslType.withNameWithDefault(dslType) else DslType.GriffinDsl
+  def getDqType: DqType.DqType = if (dqType != null) DqType.withNameWithDefault(dqType) else DqType.Unknown
```

Comment:
@tusharpatil20 setting the default values directly in the case class can eliminate the if else null check.

for example:
```
case class RuleParam(@JsonProperty("dsl.type") private val dslType: String = DslType.GriffinDsl.toString ,
....
)
```
correspondingly, 

```
def getDslType: DslType.DslType = DslType.withNameWithDefault(dslType)
```

This can be followed in other places as well. Ideally Option[T] should be used.

Reply:
I didn't change it intentionally as I am changing the existing functionality as it is.